drivers/cluster/offload_snitchCluster.c

```diff
@@ -0,0 +1,196 @@
+// Copyright 2024 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Moritz Scherer <scheremo@iis.ee.ethz.ch>
+// Philip Wiese <wiesep@iis.ee.ethz.ch>
+
+#include "soc.h"
+#include "offload_snitchCluster.h"
+
+#include <stdint.h>
+#include <stddef.h>
+
+// Persistent trampoline function pointer for each core
+void (*_trampoline_function[NUM_CLUSTER_CORES])(void *) = {NULL};
+
+// Peristent argument storage for the trampoline function
+void *_trampoline_args[NUM_CLUSTER_CORES] = {NULL};
+
+// Persistant stack pointer storage for each core
+void *_trampoline_stack[NUM_CLUSTER_CORES] = {NULL};
```

Comment:
I'd make those `static` if possible to avoid unintended tomfoolery

Reply:
FUTURE: Might also make sense (in a future PR, of course) to think about data structures for this problem - just replicating `NUM_CLUSTER_CORES` trampolines seems a bit excessive. Maybe a linked function list? function vector?